src/modbus.c

```diff
@@ -1559,6 +1728,140 @@ int modbus_report_slave_id(modbus_t *ctx, int max_dest, uint8_t *dest)
     return rc;
 }
 
+/* Read General Reference ( aka as Read File ) reads nb registers (16-bit) from
+ * an offset ( also 16-bit stepping ) of a given filenumber*/
+/* This implements only the simple case with one subrequest. More complex can be
+ * created with raw-message.  */
+/* !!! Take care, that dest must have the size of at least read_nb +1  !!! */
+
+int modbus_read_general_reference(modbus_t *ctx, int file_no, int read_addr,
+                                  int read_nb, uint16_t *dest)
```

Comment:
I would prefer "modbus_read_file_record" as name to stay as much as possible close to this document: http://www.modbus.org/docs/Modbus_Application_Protocol_V1_1b.pdf
(the Modicon naming looks somewhat "antiquated" to me)